scheduler/reserve.go

```diff
@@ -0,0 +1,128 @@
+package scheduler
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"time"
+)
+
+// ReserveAction is an interface to provide a way to reserve a resource in scheduler
+type ReserveAction interface {
+	Reserve(scheduler *Scheduler, requests []ResourceRequest, context Context, host *host, force bool, data map[string]interface{}) error
+	RollBack(scheduler *Scheduler, requests []ResourceRequest, context Context, host *host)
+}
+
+// ProvideReserveAction provides a list of reserve action. It is a gateway to register a new reserve action
+func ProvideReserveAction() []ReserveAction {
```

Comment:
ProvideReservceAction sounds odd to me. The word provide is throwing me off and it should be plural. Also, it shouldn't be capitalized since we are just using it in this package. Can you change it to either `reserveActions` or `getReserveActions`, whichever you like more